### PR TITLE
Remove AlignedArray and Aligned Matrix from src and tests

### PR DESCRIPTION
**Pull request by Anipik**
Fixes https://github.com/dotnet/machinelearning/issues/1018

cc @danmosemsft @eerhardt @TomFinley @yaeldekel @shauheen

### CODE REVIEW

File: test/Microsoft.ML.CpuMath.UnitTests.netcoreapp/UnitTests.cs [hunk 1]
```diff
@@ -555,12 +555,12 @@ public void Dist2Test(int test, float expected)
...
         {
-            AlignedArray src = new AlignedArray(8 + 8 * test, _vectorAlignment);
-            src.CopyFrom(_testSrcVectors[test]);
+            float[] src = new float[8 + 8 * test];
+            Array.Copy(_testSrcVectors[test], 0, src, 0, src.Length);
 
-            CpuMathUtils.ZeroMatrixItems(src, src.Size, src.Size, idx);
-            float[] actual = new float[src.Size];
-            src.CopyTo(actual, 0, src.Size);
+            CpuMathUtils.ZeroMatrixItems(src, src.Length, src.Length, idx);
```
Comment (eerhardt) [inline test/Microsoft.ML.CpuMath.UnitTests.netcoreapp/UnitTests.cs]: Who is calling this method besides unit tests? I don't see any. We should just delete it.

File: test/Microsoft.ML.CpuMath.UnitTests.netcoreapp/UnitTests.cs [hunk 2]
```diff
@@ -569,12 +569,12 @@ public void ZeroItemsUTest(int test, int[] idx, float[] expected)
...
         {
-            AlignedArray src = new AlignedArray(8 + 8 * test, _vectorAlignment);
-            src.CopyFrom(_testSrcVectors[test]);
+            float[] src = new float[8 + 8 * test];
+            Array.Copy(_testSrcVectors[test], 0, src, 0, src.Length);
 
-            CpuMathUtils.ZeroMatrixItems(src, src.Size / 2 - 1, src.Size / 2, idx);
-            float[] actual = new float[src.Size];
-            src.CopyTo(actual, 0, src.Size);
+            CpuMathUtils.ZeroMatrixItems(src, src.Length / 2 - 1, src.Length / 2, idx);
```
Comment (eerhardt) [inline test/Microsoft.ML.CpuMath.UnitTests.netcoreapp/UnitTests.cs]: Same for this method. I don't see any callers - it should just be removed.

**Comment by TomFinley** [comment]
Sounds great to me, thanks @eerhardt .

**Comment by Anipik** [comment]
@eerhardt @danmosemsft @TomFinley @tannergooding 
``` ini

```
|                                                            Method |    without align |    align |
|------------------------------------------------------------------ |--------:|---------:|
| Test_Multiclass_WikiDetox_BigramsAndTrichar_OVAAveragedPerceptron | 5.8   s |  5.297 s |
| CV_Multiclass_WikiDetox_BigramsAndTrichar_OVAAveragedPerceptron   | 105   s |   96.8 s |
|    CV_Multiclass_WikiDetox_BigramsAndTrichar_LightGBMMulticlass   | 310   s |   297  s |
|    CV_Multiclass_WikiDetox_WordEmbeddings_OVAAveragedPerceptron   | 310   s |   309  s |
|                   CV_Multiclass_WikiDetox_WordEmbeddings_SDCAMC   | 192   s |   196  s |
|                                                  TrainKMeansAndLR | 242   ms|   232  ms|
|                                               MakeIrisPredictions |  3   ms |   3    ms|
|                                          MakeSentimentPredictions | 51   ms |   53  ms |
|                                       MakeBreastCancerPredictions |  1.3 ms | 1.3   ms |
| Test_Ranking_MSLRWeb10K_RawNumericFeatures_FastTreeRanking        | 1.4 s   | 1.1 s    |
| TrainTest_Ranking_MSLRWeb10K_RawNumericFeatures_FastTreeRanking   | 33.91 s |  32.7 s  |
| TrainTest_Ranking_MSLRWeb10K_RawNumericFeatures_LightGBMRanking   | 34.33 s |  30 s    |

**Comment by Anipik** [comment]
The test failing is flaky.
I queued another build and it was successful. https://dnceng.visualstudio.com/public/_build/results?buildId=25891&view=results

**Comment by eerhardt** [comment]
I'm concerned with some of the perf degradation. We should investigate why these tests are that much slower.

**Comment by tannergooding** [comment]
The codegen is also going to be worse for the native intrinsic code, since it is emitting "legacy" encoded instructions (and won't be able to fold away the loads anymore).

I imagine the diff will be less for netcoreapp3.0 using the managed HWIntrinsics.

**Comment by eerhardt** [comment]
> The current code also doesn't have the fixes that ensures most of the read/writes are actually aligned (except for the few leading/trailing unaligned elements, if any)

Can we prototype that and see if that fixes the perf regression?

**Comment by markusweimer** [comment]
What is the status of this PR? I am asking because the last update has been some time ago, and it is now in conflict with `master`.

**Comment by Anipik** [comment]
@markusweimer this pr is waiting on https://github.com/dotnet/machinelearning/pull/1274 . i have the conflicts resolved in my local branch. probably #1274 will get merged by today

**Comment by Anipik** [comment]
Closing this PR, will continue the work after my vacation

**Pull request closed without merging**